Tests/tests/test_SystemServicesTest.cpp

```diff
@@ -0,0 +1,272 @@
+}
+
+TEST_F(SystemServicesTest, SystemUptime)
+{
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("requestSystemUptime"), _T("{}"), response));
+
+    size_t pos1 = response.find(",");
+
+    string result = response.substr(pos1+1, response.length()-pos1-2);
+    EXPECT_EQ(result,string("\"success\":true"));
+}
+    
+
+TEST_F(SystemServicesTest, PendingReboot)
+{
+	EXPECT_CALL(rfcApiImplMock, setRFCParameter(::testing::_, ::testing::_, ::testing::_, ::testing::_))
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [](char *pcCallerID, const char* pcParameterName, const char* pcParameterValue, DATA_TYPE eDataType) {
+                EXPECT_EQ(strcmp(pcCallerID, "thunderapi"), 0);
+                EXPECT_EQ(strcmp(pcParameterName, "Device.DeviceInfo.X_RDKCENTRAL-COM_RFC.Feature.AutoReboot.fwDelayReboot"), 0);
+		return WDMP_SUCCESS;
+	}));
+	
+	EXPECT_EQ(Core::ERROR_NONE, handlerV2.Invoke(connection, _T("fireFirmwarePendingReboot"), _T("{}"),response));
+	EXPECT_EQ(response,string("{\"success\":true}"));
+}
+
+
+TEST_F(SystemServicesTest, AutoReboot)
+{
+        EXPECT_CALL(rfcApiImplMock, setRFCParameter(::testing::_, ::testing::_, ::testing::_, ::testing::_))
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [](char *pcCallerID, const char* pcParameterName, const char* pcParameterValue, DATA_TYPE eDataType) {
+                EXPECT_EQ(strcmp(pcCallerID, "thunderapi"), 0);
+                EXPECT_EQ(strcmp(pcParameterName, "Device.DeviceInfo.X_RDKCENTRAL-COM_RFC.Feature.AutoReboot.Enable"), 0);
+                return WDMP_SUCCESS;
+        }));
+
+        EXPECT_EQ(Core::ERROR_NONE, handlerV2.Invoke(connection, _T("setFirmwareAutoReboot"), _T("{\"enable\":true}"),response));
+        EXPECT_EQ(response,string("{\"success\":true}"));
+}
+
+TEST_F(SystemServicesTest, RebootDelay)
+{
+        EXPECT_CALL(rfcApiImplMock, setRFCParameter(::testing::_, ::testing::_, ::testing::_, ::testing::_))
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [](char *pcCallerID, const char* pcParameterName, const char* pcParameterValue, DATA_TYPE eDataType) {
+                EXPECT_EQ(strcmp(pcCallerID, "thunderapi"), 0);
+                EXPECT_EQ(strcmp(pcParameterName, "Device.DeviceInfo.X_RDKCENTRAL-COM_RFC.Feature.AutoReboot.fwDelayReboot"), 0);
+                return WDMP_SUCCESS;
+        }));
+
+    EXPECT_EQ(Core::ERROR_NONE, handlerV2.Invoke(connection, _T("setFirmwareRebootDelay"), _T("{\"delaySeconds\":10}"),response));
+    EXPECT_EQ(response,string("{\"success\":true}"));
+}
+
+TEST_F(SystemServicesTest, Firmware)
+{
+	EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("updateFirmware"), _T("{}"),response));
+    EXPECT_EQ(response,string("{\"success\":true}"));
+	
+    EXPECT_EQ(Core::ERROR_NONE, handlerV2.Invoke(connection, _T("getLastFirmwareFailureReason"), _T("{}"),response));
+    EXPECT_EQ(response,string("{\"failReason\":\"None\",\"success\":true}"));
```

Comment:
according to the "/opt/fwdnldstatus.txt" you added, FailureReason is "Invalid Request". why are you testing \"None\" ?

Reply:
Fixed.